searx/engines/sqlite.py

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+# pylint: disable=missing-function-docstring
+
+"""SQLite database (Offline)
+
+"""
+
+import sqlite3
+
+engine_type = 'offline'
+database = ""
+query_str = ""
+limit = 10
+paging = True
+result_template = 'key-value.html'
+
+def init(engine_settings):
+    if 'query_str' not in engine_settings:
+        raise ValueError('query_str cannot be empty')
+
+    if not engine_settings['query_str'].lower().startswith('select '):
+        raise ValueError('only SELECT query is supported')
+
+def search(query, params):
+    query_params = {'query': query}
+    query_to_run = query_str + ' LIMIT {0} OFFSET {1}'.format(limit, (params['pageno'] - 1) * limit)
+
+    connection = sqlite3.connect(database)
+    cur = connection.cursor()
+    cur.execute(query_to_run, query_params)
+    results = _fetch_results(cur)
+    cur.close()
+    connection.close()
```

Comment:
To avoid memory / connection / cursor leak (??) :
```python
connection = sqlite3.connect(database)
try:
  cur = connection.cursor()
  try:
    cur.execute(query_to_run, query_params)
    return _fetch_results(cur)
  finally:
    cur.close()
finally:
  connection.close()
```

Reply:
> To avoid memory / connection / cursor leak (??) :

I will implement a [context manager](https://docs.python.org/3/library/stdtypes.html#context-manager-types).

Reply:
I haven't seen a context manager for sqlite except for SQL transaction:
https://docs.python.org/3/library/sqlite3.html#using-the-connection-as-a-context-manager

Reply:
> I haven't seen a context manager for sqlite

I have already implemented one .. wait until I have rebased this PR ...